src/main/java/org/opentripplanner/graph_builder/module/osm/naming/SidewalkNamer.java

```diff
@@ -0,0 +1,290 @@
+
+    LOG.info(
+      "Assigned names to {} of {} of sidewalks ({}%)",
+      namesApplied.get(),
+      unnamedSidewalks.size(),
+      DoubleUtils.roundTo2Decimals((double) namesApplied.get() / unnamedSidewalks.size() * 100)
+    );
+
+    LOG.info(progress.completeMessage());
+
+    // set the indices to null so they can be garbage-collected
+    streetEdges = null;
+    unnamedSidewalks = null;
+  }
+
+  /**
+   * Compute the centroid of all sidewalk edges.
+   */
+  private Coordinate computeEnvelopeCenter() {
+    var envelope = new Envelope();
+    unnamedSidewalks.forEach(e -> {
+      envelope.expandToInclude(e.edge.getFromVertex().getCoordinate());
+      envelope.expandToInclude(e.edge.getToVertex().getCoordinate());
+    });
+    return envelope.centre();
+  }
+
+  /**
+   * The actual worker method that runs the business logic on an individual sidewalk edge.
+   */
+  private void assignNameToSidewalk(EdgeOnLevel sidewalkOnLevel, AtomicInteger namesApplied) {
+    var sidewalk = sidewalkOnLevel.edge;
+    var buffer = preciseBuffer.preciseBuffer(sidewalk.getGeometry());
+    var sidewalkLength = SphericalDistanceLibrary.length(sidewalk.getGeometry());
+
+    var candidates = streetEdges.query(buffer.getEnvelopeInternal());
+
+    groupEdgesByName(candidates)
+      // make sure we only compare sidewalks and streets that are on the same level
+      .filter(g -> g.levels.equals(sidewalkOnLevel.levels))
+      .map(g -> computePercentInsideBuffer(g, buffer, sidewalkLength))
+      // Remove those groups where less than a certain percentage is inside the buffer around
+      // the sidewalk. This is a safety mechanism for sidewalks that snake around the corner,
+      // like https://www.openstreetmap.org/way/1059101564.
+      .filter(group -> group.percentInBuffer > MIN_PERCENT_IN_BUFFER)
+      .max(Comparator.comparingDouble(NamedEdgeGroup::percentInBuffer))
+      .ifPresent(group -> {
+        namesApplied.incrementAndGet();
+        sidewalk.setName(Objects.requireNonNull(group.name));
+      });
+  }
+
+  /**
+   * Compute the length of the group that is inside the buffer and return it as a percentage
+   * of the length of the sidewalk.
+   */
+  private static NamedEdgeGroup computePercentInsideBuffer(
+    CandidateGroup g,
+    Geometry buffer,
+    double sidewalkLength
+  ) {
+    var lengthInsideBuffer = g.intersectionLength(buffer);
+    double percentInBuffer = lengthInsideBuffer / sidewalkLength;
+    return new NamedEdgeGroup(percentInBuffer, g.name);
+  }
+
+  /**
+   * If a single street is split into several edges, each individual part of the street would potentially
+   * have a low similarity with the (longer) sidewalk. For that reason we combine them into a group
+   * and have a better basis for comparison.
+   */
+  private static Stream<CandidateGroup> groupEdgesByName(List<EdgeOnLevel> candidates) {
+    return candidates
+      .stream()
+      .collect(Collectors.groupingBy(e -> e.edge.getName()))
```

Comment:
For future work in a subsequent PR: This would not accommodate unnamed slip lanes (see screenshot and OSM link), and the adjacent sidewalk would keep its default name.

See https://www.openstreetmap.org/way/1201583893 vs. https://www.openstreetmap.org/way/784160579

<img width="311" alt="image" src="https://github.com/opentripplanner/OpenTripPlanner/assets/56846598/9ed80164-b7a4-4cb6-b5d4-73fd97886c7a">

Reply:
Well, we already filter out nameless edges berfore adding candidates to the index. Is the sidewalk in question nameless?

Reply:
Yes.